birdspotter/birdspotter/views.py

```diff
@@ -21,6 +25,27 @@ def index(request):
     return render(request, 'index.html', {'datasets': datasets})
 
 
+@login_required
+def edit_dataset(request, uuid):
+    """
+    Editing dataset metadata, such as dataset name, comments, and whether the dataset is public
+    """
+    user = request.user
+    dataset = Dataset.objects.filter(owner_id=user.id).get(dataset_id=uuid)
+    if dataset is not None:
+        form = DatasetEditForm(request.POST or None, instance=dataset)
+        if form.is_valid():
+            form.save()
+            messages.success(request, "Data set edit complete")
+            return redirect("/")
+    form = DatasetEditForm()
```

Comment:
We should provide the initial values, at least for the dataset name and the `is_public` flag. The former, for allowing a user to change a single field without having to retype/copy-paste the dataset name, and the latter to show the user what the current state is. 
```suggestion
    form = DatasetEditForm(initial={
    'name': dataset.name,
    'is_public': dataset.is_public,
    })
```

Reply:
Oh, thats really smart. Thanks for the suggestion!